### ISSUE TITLE
[CW-2005] Feature: GDPR Consent request

### ISSUE THREAD

**Issue by pranavrajs** [chatwoot/chatwoot#1231]
<img src="https://user-images.githubusercontent.com/2246121/93011698-0d0c8a00-f5b6-11ea-8e29-517ab1f4cb58.png" alt="widget-consent-form" width="400"  />

- Allow the administrator to configure a consent form on inbox settings
- Ask the user for consent in the widget
- If the user accepts it, mark that the contact has provided consent.

### Todo

- Work on a standard consent template language.

<sub>[CW-2005](https://linear.app/chatwoot/issue/CW-2005/feature-gdpr-consent-request)</sub>

**Comment by iammattmartin** [comment]
Re: Work on a standard consent template language.

I think this should be admin definable within the admin area as either a free text field with mark up or HTML editor as every entity is different as to what they need to cover.

**Comment by pranavrajs** [comment]
@iammattmartin yes, that is the idea. This will be configurable in inbox settings for each inbox. A standard consent template language should only be used as a starting point.

**Comment by mysticaltech** [comment]
Looking good... Really looking forward to that much needed feature!

**Comment by pklreak** [comment]
Another option could be this pre-chat toggle like chatbot.com does, could be easier:
![Screenshot 2021-02-12 at 16 02 00](https://user-images.githubusercontent.com/78969224/107784536-dd6fd280-6d4b-11eb-88cc-efc46fde0291.png)

**Comment by prononext** [comment]
👍 👍 👍

**Comment by uhlhosting** [comment]
How is this going on guys?

**Comment by jdabeau** [comment]
Hi! Just asking too how is it going

**Comment by Evoltstore** [comment]
Still no GDPR request :(

**Comment by jdabeau** [comment]
Hi @sojan-official 
How is it going? Thank you :)!

**Comment by tuhinmallick** [comment]
Any update on this ?

**Comment by isi2010** [comment]
@pranavrajs is chatwoot now gdpr compliant or not ?

**Comment by tuhinmallick** [comment]
@sojan-official  any update ?
